src/migration.ts

```diff
@@ -49,19 +50,108 @@ export type CreateOptions = {
 
 const SEPARATOR = '_';
 
-export async function loadMigrationFiles(
-  dir: string,
-  ignorePattern?: string
+function localeCompareStringsNumerically(a: string, b: string): number {
+  return a.localeCompare(b, undefined, {
+    usage: 'sort',
+    numeric: true,
+    sensitivity: 'variant',
+    ignorePunctuation: true,
+  });
+}
+
+function compareFileNamesByTimestamp(
+  a: string,
+  b: string,
+  logger?: Logger
+): number {
+  const aTimestamp = getNumericPrefix(a, logger);
+  const bTimestamp = getNumericPrefix(b, logger);
+
+  return aTimestamp - bTimestamp;
+}
+
+interface LoadMigrationFilesOptions {
+  /**
+   * Regex pattern for file names to ignore (ignores files starting with `.` by default).
+   * Alternatively, provide a [glob](https://www.npmjs.com/package/glob) pattern or
+   * an array of glob patterns and set `isGlob = true`
+   *
+   * Note: enabling glob will read both, `dir` _and_ `ignorePattern` as glob patterns
+   */
+  ignorePattern?: string | string[];
+  /**
+   * Use [glob](https://www.npmjs.com/package/glob) to find migration files.
+   * This will use `dir` _and_ `options.ignorePattern` to glob-search for migration files.
+   *
+   * @default: false
+   */
+  useGlob?: boolean;
+  /**
+   * Redirect messages to this logger object, rather than `console`.
+   */
+  logger?: Logger;
+}
+
+/**
+ * reads files from `dir`, sorts them and returns an array of their absolute paths.
+ * When not using globs, files are sorted by their numeric prefix values first. 17 digit numbers are interpreted as utc date and converted to the number representation of that date.
+ * Glob matches are sorted via String.localeCompare with ignored punctuation.
+ *
+ * @param dir The directory containing your migration files. This path is resolved from `cwd()`.
+ * Alternatively, provide a [glob](https://www.npmjs.com/package/glob) pattern or
+ * an array of glob patterns and set `options.useGlob = true`
+ *
+ * Note: enabling glob will read both, `dir` _and_ `options.ignorePattern` as glob patterns
+ * @param {LoadMigrationFilesOptions} options
+ * @returns {Promise<Array<String>>} Array of absolute paths
```

Comment:
**nitpick:** you don't need these `{...}` because we are in TS context

try to adjust the JSDocs a bit on what is already used in the project :)